commit ed1ce9a630b375b0f43435e34fbe690eb8276178
Author: Chun-wei Fan <fanchunwei@src.gnome.org>
Date:   Wed Nov 10 16:11:04 2021 +0800

    build: Support Visual Studio 2022
    
    Make these builds distinct from the Visual Studio 2019 builds.

diff --git a/MSVC_NMake/detectenv-msvc.mak b/MSVC_NMake/detectenv-msvc.mak
index bf2e906..58b7092 100644
--- a/MSVC_NMake/detectenv-msvc.mak
+++ b/MSVC_NMake/detectenv-msvc.mak
@@ -98,9 +98,12 @@ PDBVER = 14
 !if $(VCVERSION) > 1909 && $(VCVERSION) < 1920
 VSVER_SUFFIX = 1
 VSVER = 15
-!elseif $(VCVERSION) > 1919 && $(VCVERSION) < 2000
+!elseif $(VCVERSION) > 1919 && $(VCVERSION) < 1930
 VSVER_SUFFIX = 2
 VSVER = 16
+!elseif $(VCVERSION) > 1929 && $(VCVERSION) < 2000
+VSVER_SUFFIX = 3
+VSVER = 17
 !else
 VSVER = $(PDBVER)
 !endif
diff --git a/meson.build b/meson.build
index 8b18fc2..1fa2da0 100644
--- a/meson.build
+++ b/meson.build
@@ -238,7 +238,9 @@ if is_msvc
     add_project_arguments(disabled_warning, language: 'cpp')
   endforeach
   if use_msvc14x_toolset_ver
-    if cpp_compiler.version().version_compare('>=19.20')
+    if cpp_compiler.version().version_compare('>=19.30')
+      msvc14x_toolset_ver = '-vc143'
+    elif cpp_compiler.version().version_compare('>=19.20')
       msvc14x_toolset_ver = '-vc142'
     elif cpp_compiler.version().version_compare('>=19.10')
       msvc14x_toolset_ver = '-vc141'
